src/main/java/org/broadinstitute/hellbender/engine/FeatureContext.java

```diff
@@ -107,11 +110,77 @@ public SimpleInterval getInterval() {
      *         this FeatureContext's query interval. Empty List if there is no backing data source and/or interval.
      */
     public <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor ) {
-        if ( featureManager == null || interval == null ) {
+        return getValues(featureDescriptor, interval);
+    }
+
+    /**
+     * Gets all Features from the source represented by the provided FeatureInput argument that overlap
+     * this FeatureContext's query interval. Will return an empty List if this FeatureContext has
+     * no backing source of Features and/or interval.
+     *
+     * Returned Features are not guaranteed to be in any particular order.
```

Comment:
Add a warning to this method that if `windowLeadingBases > 0` and query lookahead caching is enabled for the underlying `FeatureDataSource`, there will be a cache miss on every call, since the current caching scheme prefetches Features after the current query interval, but not before it (on the assumption that the common access pattern involves gradually increasing query intervals).

Also create a new ticket (and assign to me) to make the caching in `FeatureDataSource` more "window friendly" (ie., prefetch and cache around the current interval instead of just after it).